engine/src/main/java/io/zeebe/engine/state/instance/ElementInstance.java

```diff
@@ -7,80 +7,95 @@
  */
 package io.zeebe.engine.state.instance;
 
-import static io.zeebe.db.impl.ZeebeDbConstants.ZB_DB_BYTE_ORDER;
-import static io.zeebe.util.buffer.BufferUtil.readIntoBuffer;
-import static io.zeebe.util.buffer.BufferUtil.writeIntoBuffer;
-
 import io.zeebe.db.DbValue;
 import io.zeebe.engine.processor.workflow.WorkflowInstanceLifecycle;
+import io.zeebe.msgpack.UnpackedObject;
+import io.zeebe.msgpack.property.IntegerProperty;
+import io.zeebe.msgpack.property.LongProperty;
+import io.zeebe.msgpack.property.ObjectProperty;
 import io.zeebe.protocol.impl.record.value.workflowinstance.WorkflowInstanceRecord;
 import io.zeebe.protocol.record.intent.WorkflowInstanceIntent;
 import org.agrona.DirectBuffer;
-import org.agrona.MutableDirectBuffer;
-
-public final class ElementInstance implements DbValue {
-
-  private final IndexedRecord elementRecord;
-
-  private long parentKey = -1;
-  private int childCount;
-  private long jobKey;
-  private int activeTokens = 0;
-
-  private int multiInstanceLoopCounter = 0;
-  private long interruptingEventKey = -1;
-
-  private long calledChildInstanceKey = -1L;
+import org.agrona.concurrent.UnsafeBuffer;
+
+public final class ElementInstance extends UnpackedObject implements DbValue {
```

Comment:
I'm not sure about this. Unless we change the db to only take msgpack format instead of DbValues, this doesn't give us much right? And we don't want to force the db values to be msgpack since in the future we might want a column family with a different format for whatever reason (even if it has different backwards compat properties). But I agree we can delete some of the DbValue implementations that are not also DbKeys.

Reply:
:+1: